### ISSUE TITLE
Session-scoped fixture that mocks requests is affecting all my tests

### ISSUE THREAD

**Issue by paxcodes** [getsentry/responses#356]
I'm not sure if this is a bug or a knowledge-gap on my part on how fixtures work.

Problem: My session-scoped fixture that mocks responses and yields within a context manager (as directed in the [README](https://github.com/getsentry/responses#responses-as-a-pytest-fixture)) is affecting a test that I didn't expect to be affected.

To Reproduce:
1. Given a session-scoped fixture that mocks responses 

tests/conftest.py
```python
@fixture(scope="session")
def stub_rss_responses():
  with responses.RequestsMock()...
    # mocking requests to RSS feeds here
    yield
```

2. And a separate test somewhere that _doesn't_ use the stub

tests/some/package/test_rss.py
```python
class TestRss:  
  def test_rss_feed(self):
    # Line below calls requests.get to retrieve the RSS feed
    actualFeed = ArticlesRssFeed()
```

Expectation: stub_rss_responses is not in the parameter list so I expected the real RSS feed will be retrieved
Actual: Mocked RSS was what was returned

**Comment by markstory** [comment]
Have you tried using `@pytest.fixture` instead? The README doesn't show using a session fixture, as they are shared across all tests run for a session. It looks like you're trying to use a standard vanilla fixture instead.

**Comment by paxcodes** [comment]
> The README doesn't show using a session fixture, as they are shared across all tests run for a session.

Right. I guess I was expecting that, if the fixture is not used by the test, `responses` will be "deactivated"...and if the fixture is used by the test, `responses` is "activated" ....somehow.

For now, I redesigned my tests so that my stub is not a session fixture, but something lower scoped (class or module) --- the stub is only used by 2 separate test modules/classes so duplication is not a big deal for now.

I initially used the session fixture so that "loading fake responses" from a file (my use case) only happens once throughout the test suite. But I imagine even that has a workaround. 

Closing this for now.
